subsys/bluetooth/audio/csis.c

```diff
@@ -920,13 +926,15 @@ int bt_csis_register(const struct bt_csis_register_param *param,
 	k_work_init(&inst->srv.work, disconnect_adv);
 #endif /* CONFIG_BT_EXT_ADV */
 
+	csis_cbs = param->cb;
+
```

Comment:
This is a temporary work around. Each CSIS instance can have their own set of callbacks, so we shouldn't use the global variable to store them. 

The callback pointer should be moved to `struct bt_csis_server`

Reply:
True.  I also think it might be a good idea to remove advertising altogether from CSIS too - but that would require some deeper design discussions.  Is this one OK to leave here for now and then take it up if/when we do broader changes?

Reply:
Yes, let's just move the callback now. 

I agree that advertising should be removed from CSIS, and then rely on just getting the service data (the RSI) from the stack via callback.

Reply:
> Yes, let's just move the callback now.

ok - moving it to bt_csis_server (and will make a new push)